src/core/qgssnappingconfig.h

```diff
@@ -52,12 +52,16 @@ class CORE_EXPORT QgsSnappingConfig
 
     /**
      * SnappingType defines on what object the snapping is performed
+     * Since QGIS 3.10 SnappingType is a flags
      */
-    enum SnappingType
+    enum SnappingType : int
     {
-      Vertex = 1, //!< On vertices only
-      VertexAndSegment = 2, //!< Both on vertices and segments
-      Segment = 3, //!< On segments only
```

Comment:
This is an API break -- you'll need to avoid this

Reply:
@nyalldawson what do you say?

Reply:
Hm tricky -- I could potentially see scripts using those integer values, yes (e.g. when saving a snapping state to xml or the like). I'd personally feel more comfortable if a whole new enum was introduced to avoid this

Reply:
The way (large parts of) our python API works right now is to expose integer values, not enum names to the developer. Resolving these to names is not straightforward and many devs don't (when serializing to xml or when reverse engineering something in scripts). Mostly because the system doesn't make it easy to do it better.
I would prefer to continue with integer support and try to get new enums in as enum classes (IIRC those are exposed as names by sip) to move devs towards using enum names instead of integer values.

Reply:
Sorry, I was away some days due to a surgery. I will try to write a new enum class SnappingTypeV2 to avoid breaking the API.

Reply:
I created a [commit](https://github.com/qgis/QGIS/pull/33622/commits/c72650e28f5c8c77e3874bbbd72fac36c8e7a91d) to introduce a new type. The new enum, is also a scopped enum (but I had a lot of trouble compiling my code to use the flags. I thought I saw some bugs in QT ping @3nids ?). 
So, It's still as a WIP.

Reply:
Can you describe the issue you have? I believe there might be an issue now if we leave the old enum in place (a certain version of SIP doesn't handle well mixed scoped and not scoped based enums in the same class). Forget about it if that brings you trouble (but please update the settings method the use the dedicated method to flags).

Reply:
The issue is that I don't have operators for flags (even when I added Q_FLAGS, Q_DECLARE_OPERATORS_FOR_FLAGS, etc.). That's why I overload [these operators](https://github.com/qgis/QGIS/pull/33622/commits/c72650e28f5c8c77e3874bbbd72fac36c8e7a91d#diff-3147d7d447000023ef8e38f46bf36fe3R363-R370)

For SIP, even with a new enum (scopped) I need to rename Vertex and Segment with a different name, so [VertexV2 and SegmentV2 here](https://github.com/qgis/QGIS/pull/33622/commits/c72650e28f5c8c77e3874bbbd72fac36c8e7a91d#diff-3147d7d447000023ef8e38f46bf36fe3R71-R72).

Reply:
@lbartoletti the reason for the new types is that we need flags?
Can we use a `Flag` (instead of a `V2`) suffix? That's self explanatory to read and will not look awkward when the V1 version is dropped in the future.

Reply:
@lbartoletti from the docs, you should use Q_FLAG and not Q_FLAGS

Reply:
> Can we use a `Flag` (instead of a `V2`) suffix? That's self explanatory to read and will not look awkward when the V1 version is dropped in the future

You're right. I'll rename it.





> from the docs, you should use Q_FLAG and not Q_FLAGS

Mmm, Maybe I have already try with Q_FLAG, but I don't remember everything I did. I've set up a new branch to test it.

Reply:
New enum (not scoped) with Q_FLAG done. (I'll polish it)